### PR TITLE
compatible: Add release_charm.yaml

### PR DESCRIPTION
**Pull request by carlcsaposs-canonical**
## Issue
[canonical/charming-actions/upload-charm](https://github.com/canonical/charming-actions/tree/main/upload-charm) does not support multiple series

[DPE-1273](https://warthogs.atlassian.net/browse/DPE-1273)

Also resolves [design bug](https://chat.canonical.com/canonical/pl/sjwya3bo3fd75ni4zbk5caqxsw)
## Solution
Replace charming-actions/upload-charm with a reusable workflow + Python script (so it's easier to maintain)

[DPE-1273]: https://warthogs.atlassian.net/browse/DPE-1273?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: release_charm.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+# (Only for Kubernetes charms) upload OCI image(s) & store revision
+oci_resources: list[OCIResource] = []
+resources = metadata_file.get("resources", {})
+for resource_name, resource in resources.items():
+    if resource["type"] != "oci-image":
+        continue
+    image_name = resource["upstream-source"]
+    logging.info(f"Downloading OCI image: {image_name}")
+    run(["docker", "pull", image_name])
```
[the remark this answers is not in the document]
Comment (taurus-forever) [reply release_charm.py:64]: My $0.05: the correct registry URL will be `ghcr.io/canonical/charmed-mysql:latest` and it is fine with me.

**Review by taurus-forever — APPROVED** [review]
That should be good enough to do the job. Nice!
Let's give it a try.

File: .github/workflows/build_charm_without_cache.yaml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        # TODO: Replace with custom image on self-hosted runner
+        run: |
+          # Copied from https://github.com/charmed-kubernetes/actions-operator/blob/96fb0b07eb675f74cf1796be812bc7e67a0d62fc/src/bootstrap/index.ts#L151
+          sudo adduser $USER lxd
+          newgrp lxd
+          sudo lxd waitready
+          sudo lxd init --auto
+          sudo iptables -F FORWARD
+          sudo iptables -P FORWARD ACCEPT
+          sudo chmod a+wr /var/snap/lxd/common/lxd/unix.socket
```
Comment (paulomach) [inline .github/workflows/build_charm_without_cache.yaml:40]: excepted line #37, is this a super safe way to enable lxd or just unnecessary/obsolete? I don't remember seeing such routine just to have LXD running

**Review by paulomach — APPROVED** [review]
minor nit, lgtm

**Pull request merged**